neutone_sdk/conv.py

```diff
@@ -0,0 +1,365 @@
+            padding = padding[0]
+        padded_kernel_size = (kernel_size - 1) * dilation
+        if padding == "valid":
+            padding_l = 0
+            padding_r = 0
+        elif padding == "same":
+            assert stride == 1, "If padding is 'same', stride must be 1"
+            if causal:
+                padding_l = padded_kernel_size
+                padding_r = 0
+            elif padded_kernel_size % 2 == 0:
+                padding_l = padded_kernel_size // 2
+                padding_r = padded_kernel_size // 2
+            else:
+                padding_l = padded_kernel_size // 2
+                # Favor right padding over left padding to match torch.nn.Conv1d
+                padding_r = padded_kernel_size // 2 + 1
+        else:
+            assert isinstance(padding, int)
+            assert padding >= 0
+            if causal:
+                padding_l = padding
+                padding_r = 0
+            else:
+                padding_l = padding
+                padding_r = padding
+
+        return padding_l, padding_r
+
+    @tr.jit.export
+    def is_cached(self) -> bool:
+        """Returns True if the convolution is cached, False otherwise."""
+        return self.cached
+
+    @tr.jit.export
+    def set_cached(self, cached: bool) -> None:
+        """
+        Sets the convolution to cached or not cached mode and resets its state.
+
+        Args:
+            cached: If True, the convolution is cached. If False, it is not cached.
+        """
+        self.cached = cached
+        # Batch size needs to be provided for TorchScript
+        self.reset(batch_size=None)
+
+    @tr.jit.export
+    def reset(self, batch_size: Optional[int] = None) -> None:
+        """
+        Resets the convolution's state. If batch_size is provided, the cached padding
+        will be resized to match the new batch size.
+
+        Args:
+            batch_size: If provided, the cached padding will be resized to match the new
+                        batch size.
+        """
+        self.padding_cached.reset(batch_size)
+
+    @tr.jit.export
+    def get_delay_samples(self) -> int:
+        """
+        Returns the number of samples that the convolution delays the output by. This
+        should always be 0 when the convolution is causal. This is ill-defined when not
+        in cached mode since the output number of samples can be different than the
+        input number of samples, so this would typically only be used in cached mode.
+        """
+        return self.padding_r
+
+    def prepare_for_inference(self) -> None:
+        """
+        Prepares the convolution for inference by disabling debug mode and ensuring the
+        convolution is in cached mode.
+        """
+        if not self.is_cached():
+            log.info(f"Converting Conv1dGeneral to cached in prepare_for_inference()")
+            self.set_cached(True)
+        self.debug_mode = False
+        self.padding_cached.prepare_for_inference()
```

Comment:
Might be unavoidable, but this should also reset it and inside this function you don't pass in the batch size anymore unlike what happens with the reset in this class.